server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -213,21 +213,38 @@ public Builder(String name, Resolution resolution, DateFormatter dateFormatter,
         }
 
         protected DateFieldType setupFieldType(BuilderContext context) {
-            DateFormatter dateTimeFormatter = DateFormatter.forPattern(format.getValue()).withLocale(locale.getValue());
-            return new DateFieldType(buildFullName(context), index.getValue(), docValues.getValue(),
-                dateTimeFormatter, resolution, meta.getValue());
+            try {
+                DateFormatter dateTimeFormatter = DateFormatter.forPattern(format.getValue()).withLocale(locale.getValue());
```

Comment:
Small comment, maybe we should scope the try/ catch down to avoid rewrapping an `IllegalArgumentException` from the constructor call ? It doesn't seem like a concern for this specific constructor, but seems like good practice.